go/vt/mysqlctl/schema.go

```diff
@@ -152,6 +104,10 @@ func (mysqld *Mysqld) GetSchema(dbName string, tables, excludeTables []string, i
 		sd.TableDefinitions = append(sd.TableDefinitions, td)
 	}
 
+	sd, err = sd.FilterTables(tables, excludeTables, includeViews)
+	if err != nil {
+		return nil, err
+	}
 	sd.GenerateSchemaVersion()
```

Comment:
FilterTables() only generates the schema version for the new copy of the schema if the original schema had a version. If no Version was set, no version will be generated.

This way FilterTables() and GenerateSchemaVersion() are independent

In this code, "sd" has no version yet and therefore FilterTables() does not set it. Therefore, it's correct to generate the schema version after FilterTables().